lib/setup.js

```diff
@@ -64,7 +64,10 @@ module.exports = function (options = {}) {
     .then(() => db.registerStorage(storage))
     .then(() => amphoraPlugins.registerPlugins(plugins))
     .then(() => { router = routes(app, providers, sessionStore); })
-    .then(() => internalBootstrap(bootstrap))
+    .then(() =>
+      db.applyLock
+        ? db.applyLock('bootstrap', internalBootstrap)
```

Comment:
Looks good, but I'd like to get some clarity on this pattern.

Reply:
This was a suggestion to avoid breaking changes:
If the storage module has an `applyLock` method, we use it.
If it doesn't, we execute the bootstrap as before (using the `bootstrap` boolean that comes from the configuration)